### DISCUSSION TITLE
偶尔读不出ASS/SSA字幕,视频文件名/字幕文件名完全一致

### DISCUSSION THREAD

**Discussion by yeezylife** [hooke007/mpv_PlayKit#85] [Q&A 问答] [answered]
就是偶尔会有一集的ASS/SSA字幕读不出来...这次前面17集都正常,18-19集读不出字幕,但是20集又没事了...
不是第一次出现这种现象了,很怪...我下了个其他的ssa字幕也不行,把字幕和视频都改名也不行
补番补到一半出去搜解决方案体验好差...

配置文件调成全默认,所有脚本全关闭,也读不出来.

解决了,和上次一样ASS转成SRT就能识别到字幕了...问题是EVA字幕组出的字幕 前面17集都能读出来 18 19 读不出是为啥

我回忆上次出现这个问题,也是发生在ASS字幕上

**Comment by hooke007** [comment, marked as answer]
这个问题经常有人问，回答：ASS编写的时候不规范，自己改规范了就行

以及这种问问题的方式。。。为什么不传所涉及的字幕样本

**Reply by yeezylife** [reply]
[18 命の選択を (H264 1440x1080 FLAC[5.1ch／2ch])[CD4F6CFB].zip](https://github.com/hooke007/MPV_lazy/files/8057282/18.H264.1440x1080.FLAC.5.1ch.2ch.CD4F6CFB.zip)
昨天补(期待已久的!)番补到一半搞的我很急,没想到传字幕...这三个文件都读不出来
其实就一两集读不出来的话转SRT也能看....

**Reply by hooke007** [reply]
第一行就不对，没必要往下看了。。。照着ass规范改吧
https://en.wikipedia.org/wiki/SubStation_Alpha#Advanced_SubStation_Alpha

**Reply by dyphire** [reply]
也算是日经问题了，libass对ass规范要求比vsfilter系严格的多，一些不符合规范的ass字幕无法被正确读取。典型的就是`[Script Info]`的缺失和大小写错误了。
常见的兼容性问题还有ass指定分辨率和实际视频分辨率不一致时libass很多标签都无法正确渲染，而vsfilter系可以正确处理。
可以说vsfilter系自开发以来就陆续兼容处理了很多错误行为，强大之余也使得字幕的制作越发不规范，隐藏了很多潜在问题。而这都会在libass上暴露出来，libass也一直在致力于完善和vsfilter系间的兼容性行为。
